test/unit/__snapshots__/switch.spec.ts.snap

```diff
@@ -1,52 +1,56 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`switch not allowed in 5.1: code 1`] = `
-"local ____exports = {}
+exports[`switch collapses empty case and minimizes conditions 1`] = `
+"require(\\"lualib_bundle\\");
+local ____exports = {}
 function ____exports.__main(self)
-    local ____switch3 = \\"abc\\"
-    goto ____switch3_end
-    ::____switch3_end::
+    local out = {}
+    repeat
+        local ____switch3 = 5
+        if ((____switch3 == 0) or (____switch3 == 1)) or (____switch3 == 2) then
+            __TS__ArrayPush(out, \\"0,1,2\\")
+        end
+        if (((____switch3 == 0) or (____switch3 == 1)) or (____switch3 == 2)) or (____switch3 == 3) then
+            __TS__ArrayPush(out, \\"3\\")
+            break
+        end
+        if not (((((____switch3 == 0) or (____switch3 == 1)) or (____switch3 == 2)) or (____switch3 == 3)) or (____switch3 == 4)) then
+            __TS__ArrayPush(out, \\"default\\")
+        end
+        if (not (((((____switch3 == 0) or (____switch3 == 1)) or (____switch3 == 2)) or (____switch3 == 3)) or (____switch3 == 4))) or (____switch3 == 4) then
+            __TS__ArrayPush(out, \\"4\\")
+        end
+    until true
+    return out
```

Comment:
@Perryvw any idea on how to get rid of all those extra parentheses?
It appears that using `lua.createBinaryExpression` is causing a benign nesting here.